UGRCodes Blog Post (#1714)
agrc/gis.utah.gov: _posts/2021-04-22-introducing-ugrcodes.md

---
title: "Introducing UGRCodes"
author:
  display_name: Erik Neemann
  email: eneemann@utah.gov
date: 2021-04-22 15:30:12
categories:
  - SGID Blog
  - Featured
tags:
  - sgid
---

> Are you tired of fighting with esoteric municipal FIPS codes? Have you even tried using them? It's not easy and it's certainly not intuitive. Five digit integer codes with minimal consistency and continuity across space, and require a lookup table to understand, don't lend themselves as a simple identifier for a municipality. As a solution, UGRC is implementing a new attribute field called the UGRCode (get it?) in our [Municipal Boundaries](https://opendata.gis.utah.gov/datasets/utah-municipal-boundaries) and [Metro Townships](https://opendata.gis.utah.gov/datasets/utah-metro-townships) data. The full list of UGRCodes can be found [here](https://docs.google.com/spreadsheets/d/19cwbXoewTynDDv_fJ3hGekD4VbLef5VxrnSJfRx8My8/edit#gid=2028111876).

Reviewer: **suggestion**: replace here with words
```suggestion
Are you tired of fighting with esoteric municipal FIPS codes? Have you even tried using them? It's not easy and it's certainly not intuitive. Five digit integer codes with minimal consistency and continuity across space, and require a lookup table to understand, don't lend themselves as a simple identifier for a municipality. As a solution, UGRC is implementing a new attribute field called the UGRCode (get it?) in our [Municipal Boundaries](https://opendata.gis.utah.gov/datasets/utah-municipal-boundaries) and [Metro Townships](https://opendata.gis.utah.gov/datasets/utah-metro-townships) data. The full list of UGRCodes can be found [in this spreadsheet](https://docs.google.com/spreadsheets/d/19cwbXoewTynDDv_fJ3hGekD4VbLef5VxrnSJfRx8My8/edit#gid=2028111876).

```

[![UGRCodes in Salt Lake County]({% link images/20210422_ugrcodes.png %} "UGRCodes in Salt Lake County")]({% link images/20210422_ugrcodes.png %}){: .flex .flex--center }

> Inspired by a suggestion from one of our colleagues at the Wasatch Front Regional Council, we believe these codes provide several advantages that make them more useful and easier to use than FIPS codes or municipality names for uniquely identifying municipalities in Utah. They also support our efforts to ensure that attribute domain values match their corresponding descriptions within our datasets. Some advantages of the UGRCodes are:

Reviewer: **question**: really!? this was bert's idea?

I thought this came up from some NG911 PSAP stuff etc. Does that deserve a mention?

Author: Yeah, he originally brought it up; I think he was using something like that on an analysis he was doing for the Wasatch Front. 
 I posed the question when I solicited a review from Nathan and Greg, but Nathan thought my reference was plenty.  I could quote retweet the IFTT tweet when it gets posted and give him a hat tip.

> Inspired by a suggestion from one of our colleagues at the Wasatch Front Regional Council, we believe these codes provide several advantages that make them more useful and easier to use than FIPS codes or municipality names for uniquely identifying municipalities in Utah. They also support our efforts to ensure that attribute domain values match their corresponding descriptions within our datasets. Some advantages of the UGRCodes are:

Reviewer: **thought**: it would be nice to link to the process we have for this since it is a little nebulous here.

> They also support our efforts to ensure that attribute domain values match their corresponding descriptions within our datasets

Author: Done

> - Much more intuitive than FIPS codes    

Reviewer: **nitpick**: remove extra whitespaces
```suggestion
- Much more intuitive than FIPS codes

```

In coming up with the 3-letter codes we tried to follow a few guiding principles, which are listed below. However, in order to deconflict the codes, add clarity, or maintain continuity, we had to stray from these principles occasionally.

- Align codes with 3-letter "City Codes" that already exist in 911 dispatch datasets wherever possible
  - E.g., Beaver (BVR), Millcreeak (MLC), Saint George (STG)
- Preferentially use the first 3 letters of a municipality's name
  - E.g., Hinckley (HIN), Layton (LAY), Mapleton (MAP)
- Incorporate the first letters of multiple-word municipality names
  - E.g., Salt Lake City (SLC), West Valley City (WVC), Bear River City (BRC)
- Distinguish municipalities with similar names or beginnings as much as possible
  - E.g., Kanab (KNB), Kanarraville (KNV), Kanosh (KNS)
- Metro Townships have "MT" in their code - or just "T" if new ones conflict with current codes
  - E.g., Emigration Canyon (EMT), Kearns (KMT), Magna (MMT)
  

> #### Future Enhancements

Reviewer: **nitpick**: it's best to have a line feed after headers
```suggestion
#### Future Enhancements

```

> In the future, we may continue to expand on this idea by including the code field in additional datasets (road centerlines, address points, etc.). We believe UGRCodes could serve as a useful and intuitive code for querying and joining data across several datasets. The expanded use of this field may depend on how useful we find the codes, your feedback as you use the codes, or more. So reach out to us and let us know what you think via email ([agrc@utah.gov](mailto:agrc@utah.gov)) or Twitter ([@maputah](https://twitter.com/maputah)).

Reviewer: **chore**: we don't allow email addresses directly in code. For reasons like the impending agrc@utah.gov to ugrc@utah.gov name change. So please switch this to use the contacts.yml file with help from the [readme](https://github.com/agrc/gis.utah.gov/blob/master/README.md#contact-information)

Author: Ahh, thanks.  I was wondering when we'd be changing the email address, too.

Author: This was a bit awkward because I couldn't find where the autopopulated text came from (hence the trial and error), but I think I eventually got it into something workable.  Although it still references "staff from AGRC".

Author: @steveoh please take another look at this PR when you get the chance

---
title: "Introducing UGRCodes"
author:
  display_name: Erik Neemann
  email: eneemann@utah.gov
date: 2021-04-22 15:30:12
categories:
  - SGID Blog
  - Featured
tags:
  - sgid
---

Are you tired of fighting with esoteric municipal FIPS codes? Have you even tried using them? It's not easy and it's certainly not intuitive. Five digit integer codes with minimal consistency and continuity across space, and require a lookup table to understand, don't lend themselves as a simple identifier for a municipality. As a solution, UGRC is implementing a new attribute field called the UGRCode (get it?) in our [Municipal Boundaries](https://opendata.gis.utah.gov/datasets/utah-municipal-boundaries) and [Metro Townships](https://opendata.gis.utah.gov/datasets/utah-metro-townships) data. The full list of UGRCodes can be found [in this spreadsheet](https://docs.google.com/spreadsheets/d/19cwbXoewTynDDv_fJ3hGekD4VbLef5VxrnSJfRx8My8/edit#gid=2028111876).

[![UGRCodes in Salt Lake County]({% link images/20210422_ugrcodes.png %} "UGRCodes in Salt Lake County")]({% link images/20210422_ugrcodes.png %}){: .flex .flex--center }

> Inspired by a suggestion from one of our colleagues at the Wasatch Front Regional Council, we believe these codes provide several advantages that make them more useful and easier to use than FIPS codes or municipality names for uniquely identifying municipalities in Utah. They also [support our efforts](https://docs.google.com/document/d/12bdtmtv-ZVF9g-lFshbobx8DbKX9LfbL44HawLjFTVw/edit) to ensure that attribute domain values match their corresponding descriptions within our datasets. Some advantages of the UGRCodes are:

Reviewer: **thought**: i was thinking of linking internally to https://gis.utah.gov/about/policy/sgid/ but that links to this doc so that seems good.

Author: yeah, I was trying to decide between the two, but the sgid page didn't have an anchor for the domain part and I figured the doc got more into the specifics on domains
